lib/auth/permissions.go

```diff
@@ -148,23 +147,24 @@ Loop:
 
 // UseSearchAsRoles extends the roles of the Checker on the current Context with
 // the set of roles the user is allowed to search as.
-func (c *Context) UseSearchAsRoles(access services.RoleGetter) error {
+func (c *Context) UseSearchAsRoles(access services.RoleGetter, clusterName string) error {
+	if len(c.Checker.GetAllowedResourceIDs()) > 0 {
+		return trace.AccessDenied("cannot search with elevated roles while already logged in with a search-based access request")
```

Comment:
I imagine this error message could make its way to a user, who might not know what a search-based access request is. Can we re-word this to make it more obvious to the user that they are already in an elevated context?

Reply:
tried to improve this a bit, but I'm not really sure what to call it other "currently logged in with a search-based access request"